app/src/main/java/com/github/epfl/meili/util/MeiliViewModel.kt

```diff
@@ -6,13 +6,13 @@ import androidx.lifecycle.ViewModel
 import com.github.epfl.meili.database.Database
 import java.util.*
 
-class MeiliViewModel<T>: ViewModel(), Observer {
+open class MeiliViewModel<T>: ViewModel(), Observer {
 
     private val mElements: MutableLiveData<Map<String, T>> = MutableLiveData()
 
-    private lateinit var database: Database<T>
+    protected lateinit var database: Database<T>
```

Comment:
One question, why are this changes being made?

Reply:
We did it so that the ForumViewModel can access it. Otherwise, we can't define the upvote/downvote functions in ForumViewModel.kt. Plus, I believe that the children of this class should be able to use this field since the added functionality would almost always be related to the database. Thank you for the question!

Reply:
Awesome! Thank you for your answer @ahmedc3ca !